Module4.1_Intro_to_Data_Structures_and_Algos/activities/Act2_HashTables/8.md

```diff
@@ -0,0 +1,19 @@
+<!--title={Hash Table as a Dictionary in Python}-->
+
+<!--badges={Algorithms:10}-->
+
+<!--concepts={Hash Tables, Dictionaries}-->
+
+# Hash Tables in Python
+
+As you now know, Hash tables are a type of data structure in which the address or the index value of the data element is generated from a hash function. That makes accessing the data faster as the index value behaves as a key for the data value. In other words Hash table stores key-value pairs but the key is generated through a hashing function.
+
+So the search and insertion function of a data element becomes much faster as the key values themselves become the index of the array which stores the data.
+
+However, when we talk about **hash tables**, we're actually talking about a type of **dictionary**. In Python, the Dictionary data types in some way, represent the implementation of hash tables. The Keys in the dictionary satisfy the following requirements.
+
+- The keys of the dictionary are hashable i.e. the are generated by hashing function which generates unique result for each unique value supplied to the hash function.
+- The order of data elements in a dictionary is not fixed.
+
+Now you maybe thinking, "Oh, so dictionaries and hashtables are basically the same thing, right?" That's not the case. The difference between the 2 is that a dictionary is generic type whereas a hashtable is the opposite. The Hashtable is a weakly typed data structure, so you can add keys and values of any Object Type to the Hashtable. The Dictionary class is a strongly types < T Key, T Value > and you must specify the data types for both the key and value. Dictionaries won't be covered extensively in this activity, but this an interesting piece of information to take note of. 
```

Comment:
Language is a little too informal here with the first sentence. I would just take it out

Reply:
I resolved this issue.

Reply:
I was referring to the paragraph above my comment, not the first paragraph. I fixed it myself but please be aware of what my comments refer to and ask questions if you're confused.